Decred Wallet/Features/TransactionDetails/TransactionDetailsViewController.swift

```diff
@@ -8,262 +8,308 @@ import UIKit
 import Dcrlibwallet
 import SafariServices
 
-class TransactionDetailsViewController: UIViewController, SFSafariViewControllerDelegate  {
-    @IBOutlet private weak var tableTransactionDetails: UITableView!    
-    @IBOutlet weak var tableView: UITableView!
-    @IBOutlet var detailsHeader: UIView!
-    @IBOutlet weak var amount: UILabel!
-    
+class TransactionDetailsViewController: UIViewController, SFSafariViewControllerDelegate {
+    @IBOutlet weak var headerView: UIView!
+    @IBOutlet weak var txTypeLabel: UILabel!
+    @IBOutlet private weak var transactionDetailsTable: SelfSizedTableView!
+    @IBOutlet weak var showOrHideDetailsBtn: UIButton!
+
     var transactionHash: String?
     var transaction: Transaction!
-    
-    var generalTxDetails: [TransactionDetails] = []
-    
+
+    var generalTxDetails: [TransactionDetail] = []
+    var isTxInputsCollapsed: Bool = true
+    var isTxOutputsCollapsed: Bool = true
+    var isTxDetailsTableViewCollapsed: Bool = true
```

Comment:
This should come before the `txInputs` and `txOutputs` collapse toggles. Not possible to toggle those without toggling first, arranging the code in a similar fashion makes it easier to visualize mentally.